dask_ml/preprocessing/data.py

```diff
@@ -915,3 +916,43 @@ def inverse_transform(self, X):
             X[col] = series
 
         return X
+
+
+class PolynomialFeatures(skdata.PolynomialFeatures):
+
+    __doc__ = skdata.PolynomialFeatures.__doc__
+
+    def fit(self, X, y=None):
+
+        self._transformer = skdata.PolynomialFeatures()
+        X_sample = X
+        if isinstance(X, dd.DataFrame):
+            X_sample = X._meta_nonempty
+        if isinstance(X, da.Array):
+            X_sample = np.ones((1, X.shape[1]), dtype=X.dtype)
+
+        # pandas dataframe treated by sklearn and returns np.array
+        self._transformer.fit(X_sample)
+        copy_learned_attributes(self._transformer, self)
+        return self
+
+    def transform(self, X, y=None):
+
+        if isinstance(X, da.Array):
+            n_cols = len(self._transformer.get_feature_names())
+            # might raise Error if columns are separated by chunks
+            chunks = (X.chunks[0], n_cols)
+            XP = X.map_blocks(self._transformer.transform, dtype=X.dtype, chunks=chunks)
+        elif isinstance(X, pd.DataFrame):
+            data = X.pipe(self._transformer.transform)
+            columns = self._transformer.get_feature_names(X.columns)
+            XP = pd.DataFrame(data=data, columns=columns)
```

Comment:
Slight preference to match scikit-learn exactly here, since this is a clone.

If you want, you could add a keyword `preserve_dataframe` to control whether we should convert the array to a dataframe.

Reply:
Since, this will force me to alter the doc-strings, I'll not be able to tackle the issue before the weekend. I think, I'll have enough time on Sunday